### PR TITLE
Use safe versions of str functions

### PR DESCRIPTION
**Pull request by Brett208**
I think we wanted to move away from using the _s versions, but they were pretty much a dropin change which was a lot easier than redesigning all these parts to use std::string and less error prone I think.

### CODE REVIEW

File: OPUNetGameSelectWnd.cpp [hunk 1]
```diff
@@ -652,7 +652,7 @@ void OPUNetGameSelectWnd::OnReceive(Packet &packet)
 		}
 		if (bTwoExternal)
 		{
-			strncat(text, "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.", sizeof(text));
+			strncat_s(text, "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.", sizeof(text));
```
Comment (DanRStevens) [inline OPUNetGameSelectWnd.cpp]: They are a drop in replacement, except when they're not. Mingw gives a compile error for this one line change. Error:
```
OPUNetGameSelectWnd.cpp:656:130: error: too few arguments to function ‘errno_t strncat_s(char*, size_t, const char*, size_t)’
```

Reference:
https://en.cppreference.com/w/c/string/byte/strncat
It seems to be selecting:
```cpp
errno_t strncat_s(char *restrict dest, rsize_t destsz, const char *restrict src, rsize_t count);
```
Not too sure why the later version of option 1 wasn't chosen:
```cpp
char *strncat( char *restrict dest, const char *restrict src, size_t count );
```

There seems to be a C++ workaround:
https://en.cppreference.com/w/cpp/string/byte/strncat
```cpp
char *strncat( char *dest, const char *src, std::size_t count );
```
Though this seems to bring back the warning with MSVC.

----

The MSDN has:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strncat-s-strncat-s-l-wcsncat-s-wcsncat-s-l-mbsncat-s-mbsncat-s-l?view=vs-2017

The relevant overload of the method being:
```cpp
template <size_t size>
errno_t strncat_s(
   char (&strDest)[size],
   const char *strSource,
   size_t count
); // C++ only
```

**Review by DanRStevens — APPROVED** [review]
I modified one change, which effectively brings back the warning for that line. It prevents hard errors with Mingw though, which didn't like the `strncat_s` change.

We can merge now without that part done, and eliminate most of the warnings. The `strncat` change needs a bit of work though.

File: OPUNetGameSelectWnd.cpp (hunk 1 above)
Comment (DanRStevens) [reply OPUNetGameSelectWnd.cpp]: Oh, I see. The first overload doesn't have the "_s". That's why it's not being selected, the name is different. It's the old method that produces the warning. The second overload has the "_s", but it also has the extra `size_t` parameter for the size of the destination buffer.

MSVC has a non-standard extension for C++, where it detects the size of fixed arrays, which the `text` variable is. Hence why MSVC can handle that one line, but standard C++ can't.

----

You can strip the `std::` prefix I added to that line so it matches the original code. We may need to hold off on solving that one warning. We may have to rewrite that one section of code.

**Comment by DanRStevens** [comment]
I'm undoing the change to `strncat` for now. I have another open branch waiting on review that touches a similar part of the code. I can build on that other change to remove the use of `strncat` here.

Edit:
I recommend we merge this now, and then review PR #45.

**Comment by Brett208** [comment]
Sounds good. Thanks for the quick review.

When working with these _s functions, I'll try to review non-MSDN documentation so that I'm not grabbing MSVC specific versions.

-Brett

**Comment by DanRStevens** [comment]
Tagging Issue #17.

----

I should probably setup a CircleCI test build using Mingw to warn about MSVC specific code.

**Pull request merged**